Lib/ntpath.py

```diff
@@ -122,54 +122,169 @@ def join(path, *paths):
-        result = splitdrive(p)
-    It is always true that:
-        result[0] + result[1] == p
-
-    If the path contained a drive letter, drive_or_unc will contain everything
-    up to and including the colon.  e.g. splitdrive("c:/dir") returns ("c:", "/dir")
-
-    If the path contained a UNC path, the drive_or_unc will contain the host name
-    and share up to but not including the fourth directory separator character.
-    e.g. splitdrive("//host/computer/dir") returns ("//host/computer", "/dir")
-
-    Paths cannot contain both a drive letter and a UNC path.
-
+    """Split path p conservatively into a drive and remaining path.
+    Returns a 2-tuple, (drive, rest). Either component may be empty.
+
+    If the source path contains a DOS drive (i.e. a letter plus a colon), the
+    remaining path is everything after the colon.
+
+    DOS drive examples:
+
+        splitdrive('C:') == ('C:', '')
+        splitdrive('C:dir') == ('C:', 'dir')
+        splitdrive('C:/') == ('C:', '/')
+        splitdrive('C:/dir') == ('C:', '/dir')
+
+    A UNC path is parsed as follows:
+
+        drive
+        vvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvv
+        "//" domain "/" junction ["/" junction] ["/" object]
+                                                ^^^^^^^^^^^^
+                                                rest
+
+    The UNC root must be exactly two separators. Other separators may be
+    repeated.
+
+    This is a generalization of the UNC specification in [MS-DTYP] 2.2.57. The
+    latter specifies the file namespace, for which the domain is referred to
+    as "host-name" (more generally "server") and the junction as "share-name".
+    The server is commonly a local or remote network name (i.e. NETBIOS name,
+    DNS name, or IP address). It can also be a non-network server provided by
+    a local redirector. The share is a resource provided by the server, such
+    as a file-system directory.
+
+    UNC drive examples in the file namespace:
+
+        splitdrive('//server/share') == ('//server/share', '')
+        splitdrive('//server///share') == ('//server///share', '')
+        splitdrive('//server/share/') == ('//server/share', '/')
+        splitdrive('//server/share/dir') == ('//server/share', '/dir')
+
+    The other supported namespace is the device namespace, which is mapped as
+    two domains, "." and "?". These domains are handled differently in some
+    contexts, such as when creating or opening a file, but for our puposes
+    here they are equivalent. In this namespace, the junction is case-
+    insensitive. Any device junction is recognized as a UNC drive, with
+    two exceptions that require additional qualification: "GLOBAL" and "UNC".
+
+    Normally the device namespace includes the local device junctions of a
+    user, such as mapped and subst drives. The "GLOBAL" junction limits this
+    view to just global devices. It must be followed either by a device
+    junction or another "GLOBAL" junction.
```

Comment:
In light of the desire to natively use `PathCchSkipRoot()`, the `splitdrive()` implementation should not support "Global". `PathCchSkipRoot()` doesn't support this prefix, and it's uncommon in practice. It's mostly used by device drivers that need to enure that they're accessing a global device when executing in an arbitrary thread context. For example, a device driver would use the NT path "\\??\\Global\\SpamDevice" instead of "\\??\\SpamDevice".